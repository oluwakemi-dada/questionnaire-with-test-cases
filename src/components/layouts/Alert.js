import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin: 0 5% 1rem 5%;
`;

const AlertBg = styled.div`
  padding: 0.8rem 2rem;
  border-radius: 0.3rem;
  background: #f7dddc;
  color: #712b29;
  margin-bottom: 2rem;
  font-size: 1.5rem;
`;

const Alert = ({ msg }) => {
  return (
    <Wrapper>
      <AlertBg>{msg}</AlertBg>
    </Wrapper>
  );
};

export default Alert;
