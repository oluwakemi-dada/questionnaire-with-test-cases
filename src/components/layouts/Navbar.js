import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { BsMoonStarsFill } from 'react-icons/bs';
import { BsSunFill } from 'react-icons/bs';

const Wrapper = styled.nav`
  height: 8rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 5rem;
  position: relative;

  @media (max-width: 768px) {
    padding: 0 3rem;
  }

  @media (max-width: 450px) {
    padding: 0 2rem;
  }

  h1 {
    font-size: 2.5rem;
    color: #5e55ef;
  }
`;

export const ThemeToggle = styled.div`
  cursor: pointer;
  color: ${(props) => (props.theme === 'light' ? '#000' : '#fff')};
`;

const Navbar = ({ theme, themeToggler }) => {
  const currTheme = localStorage.getItem('theme');

  const onClickHandler = () => {
    themeToggler();
    if (currTheme === 'light') {
      localStorage.setItem('theme', 'dark');
    } else {
      localStorage.setItem('theme', 'light');
    }
  };
  return (
    <Wrapper>
      <h1>
        <Link to='/'>QA</Link>
      </h1>
      <ThemeToggle theme={theme} onClick={onClickHandler}>
        {theme === 'light' ? <BsSunFill /> : <BsMoonStarsFill />}
      </ThemeToggle>
    </Wrapper>
  );
};

export default Navbar;
