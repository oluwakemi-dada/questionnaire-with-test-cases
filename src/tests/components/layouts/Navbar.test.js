import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../../components/layouts/Navbar';
import { mockLocalStorage } from '../../utils/mockLocalStorage';
import { ThemeToggle } from '../../../components/layouts/Navbar';

const { getItemMock, setItemMock } = mockLocalStorage();
let theme, themeToggler, wrapper;
beforeEach(() => {
  theme = 'light';
  themeToggler = jest.fn();
  wrapper = shallow(<Navbar theme={theme} themeToggler={themeToggler} />);
});

test('should render Navbar correctly with light theme', () => {
  expect(getItemMock).toHaveBeenLastCalledWith('theme');
  expect(wrapper).toMatchSnapshot();
});

test('should render Navbar correctly with dark theme', () => {
  wrapper.setProps({
    theme: 'dark',
  });
  expect(getItemMock).toHaveBeenLastCalledWith('theme');
  expect(wrapper).toMatchSnapshot();
});

test('should set theme change in localStorage on toggle from light to dark', () => {
  getItemMock.mockReturnValue('light');
  wrapper = shallow(<Navbar theme={theme} themeToggler={themeToggler} />);
  wrapper.find(ThemeToggle).simulate('click');

  expect(themeToggler).toHaveBeenCalled();
  expect(setItemMock).toHaveBeenCalledWith('theme', 'dark');
});
