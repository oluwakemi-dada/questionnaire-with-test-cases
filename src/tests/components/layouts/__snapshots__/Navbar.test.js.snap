// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar correctly with dark theme 1`] = `
<styled.nav>
  <h1>
    <Link
      to="/"
    >
      QA
    </Link>
  </h1>
  <styled.div
    onClick={[Function]}
    theme="dark"
  >
    <BsMoonStarsFill />
  </styled.div>
</styled.nav>
`;

exports[`should render Navbar correctly with light theme 1`] = `
<styled.nav>
  <h1>
    <Link
      to="/"
    >
      QA
    </Link>
  </h1>
  <styled.div
    onClick={[Function]}
    theme="light"
  >
    <BsSunFill />
  </styled.div>
</styled.nav>
`;
